### PR TITLE
Upgrade Android gradle plugin

### PR DESCRIPTION
**Pull request by tustanivsky**
This PR is supposed to address Android symbol upload issues that some Sentry self-hosted users faced previously due to the legacy gradle plugin being in use (i.e. #502).

In order to bump the Android gradle plugin version it was necessary to override engine's gradle version as well by replacing `gradle-wrapper.properties` file in build dir.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - Add setting that allows switching between the project and user directories for the internal Sentry database location on Windows/Linux ([#616](https://github.com/getsentry/sentry-unreal/pull/616))
 - Add non-ASCII characters support for user messages ([#624](https://github.com/getsentry/sentry-unreal/pull/624))
+- Upgrade Android gradle plugin ([#633](https://github.com/getsentry/sentry-unreal/pull/633))
```
Comment (bitsandfoxes) [inline CHANGELOG.md:9]: A `from X to Y` would be nice. And maybe a line about what that change made work.

File: plugin-dev/Config/FilterPlugin.ini [hunk 2]
```diff
@@ -7,10 +7,5 @@
 ;    /Extras/...
 ;    /Binaries/ThirdParty/*.dll
 /Scripts/*
-/LinuxIntermediate/Build/Linux/UnrealEditor/Inc/Sentry/UHT/*
-/LinuxIntermediate/Build/Linux/UnrealGame/Development/Sentry/*
-/LinuxIntermediate/Build/Linux/UnrealGame/Inc/Sentry/UHT/*
-/LinuxIntermediate/Build/Linux/UnrealGame/Shipping/Sentry/*
-/LinuxIntermediate/Build/Linux/x64/UnrealGame/Development/Sentry/*
-/LinuxIntermediate/Build/Linux/x64/UnrealGame/Shipping/Sentry/*
+/Gradle/*
```
Comment (bitsandfoxes) [inline plugin-dev/Config/FilterPlugin.ini:10]: Is that change intentional? The how do the `LinuxIntermediate` related to `Gradle`?

File: plugin-dev/Source/Sentry/Sentry_Android_UPL.xml [hunk 3]
```diff
@@ -137,7 +141,7 @@
     <buildscriptGradleAdditions>
         <insert>
             dependencies {
-                classpath 'com.android.tools.build:gradle:3.5.4'
+                classpath 'com.android.tools.build:gradle:7.4.2'
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Sentry_Android_UPL.xml:144]: That's the gradle tool itself, right? And that's a requirement for the sentry-android-gradle-plugin `4.11.0`?
Are we going to mess with user's build by messing with the gradle version that comes with the UE installation?

Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Sentry_Android_UPL.xml:144]: Yes, the `io.sentry.android.gradle >= 3.0.0` requires `Android Gradle Plugin >= 7.0.0` according to docs.
I've tested this with a few different engine versions (including pretty old UE 4.27) and the build was successful however there's always a chance that some compatibility issues with 3rd-party plugins can arise.

File: plugin-dev/Config/FilterPlugin.ini (hunk 2 above)
Comment (tustanivsky) [reply plugin-dev/Config/FilterPlugin.ini:10]: Yes, aside from adding `Gradle` here I've also removed `LinuxIntermediate` stuff which is now redundant (probably I've missed deleting these entries earlier). Should we tackle this kind of clean-up in a separate PR?

File: plugin-dev/Source/Sentry/Sentry_Android_UPL.xml [hunk 4]
```diff
@@ -134,22 +152,30 @@
...
+                        <log text="Using legacy Sentry Gradle plugin 2.1.5 for Android debug symbol upload."/>
+                        <insert>
+                            dependencies {
+                                classpath 'com.android.tools.build:gradle:3.5.4'
+                                classpath 'io.sentry:sentry-android-gradle-plugin:2.1.5'
+                            }
+                        </insert>
+                    </true>
+                    <false>
+                        <log text="Using Sentry Gradle plugin 4.11.0 for Android debug symbol upload."/>
```
Comment (bitsandfoxes) [inline plugin-dev/Source/Sentry/Sentry_Android_UPL.xml:169]: Is the hardcoded version number going to bite us in the butt later on?

**Review by bitsandfoxes — COMMENTED** [review]
I'm not sure I see how this fixes the issue linked. The issue was resolved by adding the `defaults url`, right? How does the newer Android Gradle Plugin help with this? Is there anything else we're getting out of it?

The fallback is great, I really appreciate it, changing the user's Gradle version still makes me nervous.

**Comment by tustanivsky** [comment]
> I'm not sure I see how this fixes the issue linked. The issue was resolved by adding the defaults url, right? How does the newer Android Gradle Plugin help with this? Is there anything else we're getting out of it?

With the newer Gradle plugin version adding `defaults.url` to `sentry.properties` when using Sentry self-hosted shouldn't be required any longer. It was a legacy issue but now the right URL can be extracted from the authentication token so users won't face the problem with uploading symbols.

File: plugin-dev/Source/Sentry/Sentry_Android_UPL.xml (hunk 4 above)
Comment (tustanivsky) [reply plugin-dev/Source/Sentry/Sentry_Android_UPL.xml:169]: Since we don't have any means allowing to check how Sentry Gradle plugin performs in our CI probably it'd be safer to keep a hardcoded version here instead of bumping it automatically for the time being.

**Review by bitsandfoxes — APPROVED** [review]
Considering that we cannot offer ootb support for self-hosted and there is a fallback/opt-out for those that need it: LGTM! 🔥

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 - Add setting that allows switching between the project and user directories for the internal Sentry database location on Windows/Linux ([#616](https://github.com/getsentry/sentry-unreal/pull/616))
 - Add non-ASCII characters support for user messages ([#624](https://github.com/getsentry/sentry-unreal/pull/624))
+- Upgrade Android gradle plugin from v2.1.5 to v4.11.0 ([#633](https://github.com/getsentry/sentry-unreal/pull/633))
```
Comment (bruno-garcia) [inline CHANGELOG.md:9]: @tustanivsky @bitsandfoxes would this require a major bump of this SDK?

Or since we're still experimental (`0.x`) at least a callout in the changelog larger than this one liner?

Comment (tustanivsky) [reply CHANGELOG.md:9]: This isn't a breaking change so a major version bump is not required. However it'll make sense to mention in docs that there's a fallback allowing users to switch back to the legacy gradle plugin in case any issues occur.
